lib/internal/perf/timerify.js

```diff
@@ -75,18 +74,17 @@ function timerify(fn, options = {}) {
     const result = constructor ?
       ReflectConstruct(fn, args, fn) :
       ReflectApply(fn, this, args);
-    if (!constructor && typeof result?.finally === 'function') {
-      return result.finally(
-        FunctionPrototypeBind(
-          processComplete,
-          result,
-          fn.name,
-          start,
-          args,
-          histogram));
+    if (!constructor && typeof result?.then === 'function') {
+      return result.then(
+        function wrappedTimerifiedPromise(value) {
+          processComplete(fn.name, start, args, histogram);
```

Comment:
One idea which is not relate to the bug directly, but:

currently, `now` is called in the `processComplete` (`duration = now() - start`) , but consider the goal is to calc the time as precise as possible, it may be better to calc it outside like:
```js
const t = now()
processComplete(fn.name, start - t, args, histogram)
```
(also need to modify other related places, omit here)

This could rule out the invoking time of `processComplete`, in the cases of microbenchmark, such cost can't be ignored.